src/hotspot/share/runtime/sharedRuntime.cpp

```diff
@@ -2704,6 +2705,7 @@ void AdapterHandlerLibrary::initialize() {
   AdapterBlob* obj_int_arg_blob = nullptr;
   AdapterBlob* obj_obj_arg_blob = nullptr;
   {
+    _adapter_handler_table = new (mtClass) AHTable();
```

Comment:
This should be mtCode not mtClass since the adapter_handler_table belongs to the mtCode (NMT code tracking category).